### DISCUSSION TITLE
[Feature] Allow for input prior to generating PDF

### DISCUSSION THREAD

**Discussion by diderich** [YetiForceCompany/YetiForceCRM#14193] [Feature improvement]
### Describe the solution you'd like

The PDF functionality allows for designing highly customized reports which is great.

I have designed reports to list all time tracker entries associated with a given user. I would like to be able to produce these reports for a specific month and/or a specific year, or even better between two dates that the user can enter.

it would be great if the PDF functionality would optionally allow for the user to enter one or more values (of various uitypes) that would then be available in the .php files used to generate the reports ($(custom: ReportName)$).

### Describe alternatives you've considered

Currently I have created multiple reports, one for each possible input parameter:

TimetrackerPerUserMonthCurrent
TimetrackerPerUserMonthPrevious
TimetrackerPerUserYearCurrent
TimetrackerPerUserYearPrevious

Having additional variables entered by the user, I could reduce the number of reports to one and generate more flexible reports, by allowing the user to enter a start and an end date.

### Additional context

Here is a sample report I designed (TimetrackerPerUserMonthPrevious):

![report](https://user-images.githubusercontent.com/10230144/78577917-14717f80-782f-11ea-81ae-363f3fb0a6df.png)

I would potentially be willing to pay for such a feature to be developed, if the cost is reasonable.

**Comment by waran70** [comment]
Kinda of missing reporting replacement. Good idea.
